### PR TITLE
25th Bug Crusade

### LINKED ISSUE #1068: Ymiron Laws
**Your mod version is:**
Master

**What expansions do you have installed?**
All
**Are you using any submods/mods? If so, which?**
none
**Please explain your issue in as much detail as possible:**
In the laws it changed to void magic and full status for woman 
**Steps to reproduce the issue:**
Awaken Ymiron then wait a few months then check the laws, this may take some time happen.
**Upload an attachment below: .zip of your save, or screenshots:**
image
image

### LINKED ISSUE #1070: rival_skull desc
**Mod Version**
b34d09e2

**Please explain your issue in as much detail as possible:**
 This may be not a human skull

**Upload screenshots of the problem localization:**

<details>
<summary>Click to expand</summary>

image
image

</details>

### LINKED ISSUE #1071: Burning your own children [Exploit]
**Your mod version is:**
b34d09e2

**What expansions do you have installed?**
All

**Are you using any submods/mods? If so, which?**
Guardians of Azeroth

**Please explain your issue in as much detail as possible:**
1.) You can kill your own children via "burn evil being" decision after they convert to evil religion from guardian. This can be used to avoid problems with gavelkind, etc.

2.) You don't get kinslayer trait from it.

**Steps to reproduce the issue:**
Assign your child a guardian with evil religion.

**Upload an attachment below: .zip of your save, or screenshots:**

<details>
<summary>Click to expand</summary>

image
image

</details>

### LINKED ISSUE #1080: fort buildings has no loc
**Mod Version**
56f4e2d9

**Are you using any submods/mods? If so, which?**
Guardians of Azeroth

**Please explain your issue in as much detail as possible:**
Looks like you moved something that you should't in #1062

**Upload screenshots of the problem localization:**

<details>
<summary>Click to expand</summary>

image

</details>

### LINKED ISSUE #1106: Request Army from Yogg-Saron
**Your mod version is:**
1.6.0
**What expansions do you have installed?**
All
**Are you using any submods/mods? If so, which?**
Yes, some submods I made myself. They don't, however, have any effect on this bug.
**Please explain your issue in as much detail as possible:**
Under the request_army_offmap decision, in the allow block, the trigger for Yogg-Saron is missing an underscore between yogg and saron. It's written as yoggsaron_offmap_decisions_enabled_trigger = yes when it should be yogg_saron_offmap_decisions_enabled_trigger = yes. As a result, this prevents the decision from being able to be taken despite meeting the requirements. The same issues exists for the request powers decision in the same file.
**Steps to reproduce the issue:**
Take a look in the wc_offmap_decisions.txt file
**Upload an attachment below: .zip of your save, or screenshots:**

### PR DESCRIPTION
**Pull request by zumbak04**
## Changelog:
- **Burn Evil Being** and **Burn Evil Beings** decisions add kinslayer traits if you burn your kin. Fixes #1071.
- Fixed #1106 issue where you couldn't use Yogg-Saron's powers.

## Developer changelog:
- Fixed #1080 issue where fort buildings had no localization.
- Fixed #1068 issue where Ymiron had Full Status of Woman, Void Magic Allowed laws after the reawakening.
- Fixed #1070 issue where **Skull Trophy** artifact mentioned that it's a human skull.

## Error changelog:
- Fixed `persistent.cpp:40` error from `error.log`.
- Fixed `effectimplementation.cpp:2030` error from `error.log`.
- Fixed `triggerimplementation.cpp:9541` error from `error.log`.
- Fixed `effectimplementation.cpp:22757` error from `error.log`.
- Fixed `effectimplementation.cpp:22816` error from `error.log`.
- Fixed `effectimplementation.cpp:24396` error from `error.log`.
- Added origin descriptions for mod non-aggression pacts. This fixes `effectimplementation.cpp:2030` error from `error.log`.
- Fixed typo where inside a chance modifier was `trait` instead of `factor`. This fixes `triggerimplementation.cpp:9541` error from error.log.

## Script changelog (ignore if you can't read CK2 script):
- Updated `terrain.txt` to the current vanilla version.
- Remove unused `inland_Sea` modifier from `terrain.txt`.
- Replaced not existing trait `undead` with `being_undead`.
- Fixed that `EXCELLENT` option inside of Felwell event didn't have any text.
- The Twilight Hour event now properly has a 33% chance to give your character lunatic trait instead of 1%.
- Fixed that some of `add_trait` effects in traditions events were inside of `limit` block.
- Commented Zoroastrian pets events in on_action.
- Fixed that Cthraxxi generals from offmap events didn't receive class traits.
- Fixed that some triggers in polygamy events didn't work because they were inside of a limit block.

# How to test:
First, you need to find out how to turn on [Log Files](https://ck2.paradoxwikis.com/Troubleshooting#Log_files).
Then open the game, start any character, close the game, open `error.log` and try to find the errors I mentioned. If you don't find any, well, it means they're fixed... yay!

Also, it would be cool if you can read the CK2 script and take a look at **_Files changed_** above.

To find more info about the issue... including screenshots, click on `#<Issue ID>`... if there's any.

### CODE REVIEW

File: localisation/00_wc_vanilla_replacements.csv [hunk 1]
```diff
@@ -153,4 +153,5 @@ HF_222_DESC;My servants apparently lost track of [child_of_destiny.GetFirstName]
 HF_222_OPTA;[child_of_destiny.GetFirstName], oh how you scared me...! And how do you know Kalimag?;;[child_of_destiny.GetFirstName], oh wie habt Ihr mich erschreckt...! Und woher kennt Ihr Kalimag?;;;;;;;;;;;x
 child_of_mithridates;Child of Shadra;;Kind von Shadra;;;;;;;;;;;x
 child_of_mithridates_desc;This character resists poison to such a degree that people claim [GetHerHim] a child of Shadra.;;Dieser Charakter widersteht Gift in einem solchen Ma�e, dass die Leute behaupten, [GetSheHe] sei ein Kind von Shadra.;;;;;;;;;;;x
-bloodline_desc_child_of_destiny;A [child_of_destiny.GetManWoman] of true legend. [child_of_destiny.GetHerHisCap] exploits mimicked those of the ancient heroes!;;Ein wahres [child_of_destiny.GetManWoman] der Legenden. [child_of_destiny.GetSheHeCap] eifert den Heldentaten der alten Helden nach!;;;;;;;;;;;x
\ No newline at end of file
+bloodline_desc_child_of_destiny;A [child_of_destiny.GetManWoman] of true legend. [child_of_destiny.GetHerHisCap] exploits mimicked those of the ancient heroes!;;Ein wahres [child_of_destiny.GetManWoman] der Legenden. [child_of_destiny.GetSheHeCap] eifert den Heldentaten der alten Helden nach!;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/00_wc_vanilla_replacements.csv]:
```suggestion
bloodline_desc_child_of_destiny;A [child_of_destiny.GetManWoman] of true legend. [child_of_destiny.GetHerHisCap] exploits mimicked those of ancient heroes!;;Ein wahres [child_of_destiny.GetManWoman] der Legenden. [child_of_destiny.GetSheHeCap] eifert den Heldentaten der alten Helden nach!;;;;;;;;;;;x
```

File: localisation/00_zuwc_united_localisation.csv [hunk 2]
```diff
@@ -1693,7 +1693,7 @@ EVTOPTA_WCFEL_1005;The Qiraji will devour them all!;;Die Qiraji werden sie alle
 EVTOPTB_WCFEL_1005;The ancient evil must remain in confinement!;;Das uralte B�se muss eingesperrt bleiben!;;;;;;;;;;;x
 burn_evil_beings;Burn Evil Beings;;Verbrennt das B�se;;;;;;;;;;;x
 burn_evil_beings_desc;Burn all unholy characters as evil beings or get a reason to execute them if they're rulers. They deserve to die!;;Verbrennt alle unheiligen Wesen. Sie verdienen nur den Tod!;;;;;;;;;;;x
-burn_evil_being_effect_tooltip;�Y[This.GetBestName]�! is sentenced to be burnt\n;;�Y[This.GetBestName]�! wird zum Tod durch Verbrennung verurteilt.;;;;;;;;;;;x
+burn_evil_being_effect_tooltip;�Y[This.GetBestName]�! is sentenced to be burnt;;�Y[This.GetBestName]�! wird zum Tod durch Verbrennung verurteilt.;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/00_zuwc_united_localisation.csv]: Not necessarily a correction, but burnt is just more often used as an adjective. "Burnt food"
```suggestion
burn_evil_being_effect_tooltip;�Y[This.GetBestName]�! is sentenced to be burned;;�Y[This.GetBestName]�! wird zum Tod durch Verbrennung verurteilt.;;;;;;;;;;;x
```

**Comment by zumbak04** [comment]
> When I tried, persistent.cpp:40 and triggerimplementation.cpp:9541 appeared. The others seem to be fixed.

@LordSilvermane fixed them in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/commit/ff65d9504c856392690ad899ee1f5802c2636a43.

**Comment by zumbak04** [comment]
> Still seeing persistent.cpp:40. However, unlike before, it only appears once. triggerimplementation.cpp:9541 was fixed.

Can't confirm. I don't see it in my log.
[error.log](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/files/4210945/error.log)
Send me yours, please.

**Comment by zumbak04** [comment]
Checked files. The PR seems ready to merge.

**Pull request merged**